### ISSUE TITLE
I do not know why placeholder can not init  meet [Graph tensors must be fetched by running a session]

### ISSUE THREAD

**Issue by mullerhai** [tensorflow/java#418]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04 x86\_64): mac  latest
- TensorFlow installed from (source or binary): yes 
- TensorFlow version (use command below):  2.4
- Java version (i.e., the output of `java -version`):jdk 15
- Java command line flags (e.g., GC parameters): not sure
- Python version (if transferring a model trained in Python): 3.8
- Bazel version (if compiling from source): not sure 
- GCC/Compiler version (if compiling from source): not sur
- CUDA/cuDNN version: no
- GPU model and memory: no
- 

**Describe the current behavior**

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

```
import org.tensorflow.framework.losses.MeanAbsolutePercentageError
import org.tensorflow.ndarray.FloatNdArray
import org.tensorflow.op.{Op, Ops}
import org.tensorflow.proto.framework.DataType
import org.tensorflow.types.{TFloat32, TFloat64, TInt32, TInt64}
import org.tensorflow.types.family.{TNumber, TType}
import org.tensorflow.{EagerSession, Graph, Operand, RawTensor, Session, Tensor}
import org.tensorflow.ndarray.Shape
import org.tensorflow.op.core.Placeholder
import org.tensorflow.utils.TestSession

import java.io.{OutputStreamWriter, PrintWriter}

  def varPlaceHolder(): Unit ={
    val g = new Graph()
    val ops = Ops.create(g)
    val init = ops.withInitScope
    val a = ops.variable(ops.constant(3.0))
    val logitsArray: Array[Float] = Array(100.0f, -100.0f, -100.0f, -100.0f, 100.0f, -100.0f, -100.0f, -100.0f, 10.0f,100.0f, -100.0f, -100.0f, -100.0f, 100.0f, -100.0f, -100.0f, -100.0f, 100.0f,
  100.0f, -100.0f, -100.0f, -100.0f, 100.0f, -100.0f, -100.0f, -100.0f, 100.0f,
    100.0f, -100.0f, -100.0f, -100.0f, 100.0f, -100.0f, -100.0f, -100.0f, 100.0f)
    val logits = ops.reshape(ops.constant(logitsArray), ops.constant(Shape.of(6, 6)))
    val b = ops.variable(ops.random.randomStandardNormal( ops.constant(Array(6, 6)),classOf[TFloat64]))
    val x = ops.placeholder(classOf[TFloat64])
    val y = ops.placeholder( classOf[TFloat64],Placeholder.shape(Shape.of(6, 6)))
    val ax = ops.math.mul(a, x)
    val by = ops.math.mul(b, y)
    val z = ops.math.add(ax, by)
    val session = new Session(g)

      session.initialize()
//    val rad =ops.random.randomStandardNormal( ops.constant(Array(6, 6)),classOf[TFloat32])
//)// ops.random.randomStandardNormal( ops.constant(Array(6, 6)),classOf[TFloat32]).asTensor()  

    val res = session.runner()
      .feed(x, TFloat64.scalarOf(3.0))
      .feed(y, logits.asTensor()  ) //here meet error
      .fetch(z)
      .run
      .get(0)

    res match {
      case tf: TFloat64 => println(tf.getDouble()) // 3*3 + 2*6 = 21
      case other => println(s"OTHER: $other")
    }
    session.close()
  }
```
the error log
```
2022-02-15 12:19:46.716297: I external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Exception in thread "main" java.lang.IllegalStateException: Graph tensors must be fetched by running a session
	at org.tensorflow.GraphOperation.tensor(GraphOperation.java:187)
	at org.tensorflow.Output.asTensor(Output.java:97)
	at org.tensorflow.Operand.asTensor(Operand.java:65)
	at org.tensorflow.ops.GraphTest$.varPlaceHolder(GraphTest.scala:59)
	at org.tensorflow.ops.GraphTest$.main(GraphTest.scala:31)
	at org.tensorflow.ops.GraphTest.main(GraphTest.scala)

```

**Comment by Craigacp** [comment]
The logits variable isn't a tensor, it's a series of graph operations which can produce a tensor as an output. If you want to feed the logits array in you can use one of the `TFloat32.tensorOf` methods and supply a TensorFlow ndarray as input (which you can convert a regular array into).

**Comment by karllessard** [comment]
I’d add to this that the `asTensor()` method only works in eager mode, where tensors could be fetch after each operation. 

In your case, `logits` is the result of a graph computation (since `ops` was created with `Ops.create(g)`) and in such case, tensors can only be retrieved and manipulated from Java after they are fetched by graph session (via the `.fetch(…)` methods of a session runner). That what the error message actually means.

**Comment by mullerhai** [comment]
> I’d add to this that the `asTensor()` method only works in eager mode, where tensors could be fetch after each operation.
> 
> In your case, `logits` is the result of a graph computation (since `ops` was created with `Ops.create(g)`) and in such case, tensors can only be retrieved and manipulated from Java after they are fetched by graph session (via the `.fetch(…)` methods of a session runner). That what the error message actually means.

thank you it is work
